logicrunner/builtin/contract/rootdomain/rootdomain.go

```diff
@@ -159,30 +173,45 @@ func (rd *RootDomain) AddBurnAddress(burnAddress string) error {
 	return nil
 }
 
-// AddNewMemberToMaps adds new member to PublicKeyMap and BurnAddressMap.
-func (rd *RootDomain) AddNewMemberToMaps(publicKey string, burnAddress string, memberRef insolar.Reference) error {
+// AddNewMemberToMaps adds new member to PublicKeyMap and MigrationAddressMap.
+func (rd *RootDomain) AddNewMemberToMaps(publicKey string, migrationAddress string, memberRef insolar.Reference) error {
 	trimmedPublicKey := trimPublicKey(publicKey)
-	if _, ok := rd.PublicKeyMap[trimmedPublicKey]; ok {
-		return fmt.Errorf("member for this publicKey already exist")
+	i := foundation.GetShardIndex(trimmedPublicKey, insolar.GenesisAmountPublicKeyShards)
+	if i >= len(rd.PublicKeyShards) {
```

Comment:
why do we need this check?

Reply:
Because maybe sometime we might break GetShardIndex and it is better to check in contract that all is correct to avoid panic